src/plugins/formActions/CreateActionSpec.js

```diff
@@ -0,0 +1,653 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2022, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import CreateAction from './CreateAction';
+
+import {
+    createOpenMct,
+    resetApplicationState
+} from 'utils/testing';
+
+import { debounce } from 'lodash';
+
+let parentObject;
+let parentObjectPath;
+
+describe("The create action plugin", () => {
+    let openmct;
+
+    beforeEach((done) => {
+        openmct = createOpenMct();
+
+        openmct.on('start', done);
+        openmct.startHeadless();
+    });
+
+    afterEach(() => {
+        return resetApplicationState(openmct);
+    });
+
+    describe('creates new objects for a', () => {
+        beforeEach(() => {
+            parentObject = {
+                name: 'mock folder',
+                type: 'folder',
+                identifier: {
+                    key: 'mock-folder',
+                    namespace: ''
+                },
+                composition: []
+            };
+            parentObjectPath = [parentObject];
+
+            spyOn(openmct.objects, 'save');
+            openmct.objects.save.and.callThrough();
+
+            spyOn(openmct.forms, 'showForm');
+            openmct.forms.showForm.and.callFake(formStructure => {
+                return Promise.resolve({
+                    name: 'test',
+                    notes: 'test notes',
+                    location: parentObjectPath
+                });
+            });
+        });
+
+        afterEach(() => {
+            parentObject = null;
+        });
+
+        it('type clock', (done) => {
```

Comment:
I think we can future proof this by looping through open mct types `for(let type in openmct.types.types)` and dynamically create the it statements here, ex: ``it(`type ${type}`, (done) => ``, this way anytime new types are added they're automatically checked.

Reply:
I agree, great suggestion. My only tweak would be to use `openmct.types.listKeys()` as it's public API.

Reply:
hmm problem is anything out side `beforeEach`, `it` or `afterEach` run before (under `describe`)  even we reach this test suite. so `openmct` is `undefined` and cant add dynamic `it` statements.

Reply:
I think I may have a "solution", since you're manually including the types anyways, let's put them into an array outside of the tests, (ie. `const types = ['clock', 'table', 'etc...']`). Then you can loop through those I believe without issue: `
types.forEach((type) => { it('type ${type}', ...});
`

Then have one other test, that will check that we're covering all types:
 `expect(openmct.types.listKeys().length).toEqual(types.length);`

Then we'll know if we need to add any new types when they are added.